### PR TITLE
Speed up queries for grade editor.

### PR DESCRIPTION
**Pull request by spark008**
Loading the editor before: 1423ms (1260.6ms Views, 131.4ms ActiveRecord)
Loading the editor now: 589ms (532.2ms Views, 8.6ms ActiveRecord)

I think another area for speedup could be in loading/caching the partial that gets rendered for each grade. I tried rewriting the views to use collections, but I didn't see a speedup. There's a gem for speeding up partial rendering for collections, but the views became more complicated in order to accommodate collections, so maybe that's for another time.

### CODE REVIEW

**Comment by spark008** [comment]
Btw, I tested the change against seed data.

File: app/controllers/grades_controller.rb [hunk 1]
```diff
@@ -55,6 +57,17 @@ def editor
     end
 
     @metrics = @assignment.metrics
+    @grades = {}
+    @metrics.each do |m|
```
Comment (pwnall) [inline app/controllers/grades_controller.rb]: `m` -> `metric`, `g` -> `grade`

File: app/controllers/grades_controller.rb [hunk 2]
```diff
@@ -40,6 +40,8 @@ def index
   # GET /6.006/grades/editor
   def editor
     @subjects = current_course.students.includes(:profile).sort_by &:name
+    @registrations = {}
+    current_course.registrations.each { |r| @registrations[r.user_id] = r }
```
Comment (pwnall) [inline app/controllers/grades_controller.rb]: `r` -> `registration`

**Comment by pwnall** [comment]
Please fix the nits and merge. I'm sure the course staff people will be happier without knowing why :smile:

**Comment by spark008** [comment]
Updated!

**Comment by pwnall** [comment]
Thank you!

**Pull request merged**